numpy/distutils/ccompiler.py

```diff
@@ -136,20 +137,54 @@ def CCompiler_spawn(self, cmd, display=None):
         if is_sequence(display):
             display = ' '.join(list(display))
     log.info(display)
-    s, o = exec_command(cmd)
-    if s:
-        if is_sequence(cmd):
-            cmd = ' '.join(list(cmd))
-        try:
-            print(o)
-        except UnicodeError:
-            # When installing through pip, `o` can contain non-ascii chars
-            pass
-        if re.search('Too many open files', o):
-            msg = '\nTry rerunning setup command until build succeeds.'
-        else:
-            msg = ''
-        raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
+    try:
+        # cmd is documented above to be of type str, but
+        # check_output wants a list -- we use split() to
+        # produce a list irrespective of whether cmd is a single
+        # or multiple commands; historically, exec_command() could
+        # handle string input directly
+        if not is_sequence(cmd):
+            cmd = cmd.split(' ')
+        o = subprocess.check_output(cmd)
+    except subprocess.CalledProcessError as exc:
+        o = exc.output
+        s = exc.returncode
+    except (OSError, AttributeError):
```

Comment:
I think I can remove the `AttributeError` catch now that  `split()` is guarded by the `is_sequence` check. The latter `is_sequence` check seems like something the standard library may also have something suitable for, though I should perhaps confine refactoring to essential stuff